homework/index.js

```diff
@@ -30,15 +33,189 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+  const root = document.getElementById('root');
+  const container = createAndAppend('div', root, { id: 'container' });
+
+  function getSelectedData(value, data) {
+    let selected;
```

Comment:
If you look carefully, the operation which you are trying to do here is a `filter` operation i.e out of `data`, you want to filter out the entry where the name matches `value` being passed. Can you try to use the filter operation here instead? 😉

Reply:
Hi Ufuk - this is another way how you can write the `getSelectedData` method using the Array.prototype.filter method(https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter).

```
function getSelectedData(value, data) {
  return data.filter(repo => repo.name === value)[0];
}
```

Reply:
Even better, you can try to use `Array.prototype.find` method here. More information can be found here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

This way, you need not return the first element from the filter operation and can re-write the function as:
```
function getSelectedData(value, data) {
  return data.find(repo => repo.name === value);
}
```